spec/pubid_iso/renderer/base_spec.rb

```diff
@@ -0,0 +1,124 @@
+module Pubid::Iso
+  RSpec.describe Renderer::Base do
+    subject { described_class.new(params) }
+
+    describe "#prerender" do
+      context "attribute type_stage" do
+        subject { described_class.new(params).prerender(pdf_format: pdf_format).prerendered_params[:type_stage] }
+        let(:params) { { stage: stage, type: type, copublisher: copublisher } }
+        let(:copublisher) { nil }
+        let(:type) { nil }
+        let(:stage) { nil }
+        let(:pdf_format) { false }
+
+        context "when type and stage" do
+          let(:stage) { Stage.new(abbr: :WD) }
+          let(:type) { "TR" }
+
+          context "when no copublisher" do
+            it { is_expected.to eq("/WD TR") }
+
+            context "when pdf format" do
+              let(:pdf_format) { true }
+              it { is_expected.to eq("WD TR") }
+            end
+          end
+
+          context "when have copublisher" do
+            let(:copublisher) { "IEC" }
+
+            it { is_expected.to eq(" TR WD") }
```

Comment:
When does "TR WD" happen?

Reply:
> When does "TR WD" happen?

I cannot find real identifiers with "TR WD" but with the same pattern we have "ISO/IEC TR CD 20943-5" made from original identifier "ISO/IEC PDTR 20943-5". (https://github.com/metanorma/pubid-iso/blob/716857e92d7462f734276ade83d55caa4cc03efb/spec/pubid_iso/identifier/identifier_spec.rb#L328)

Reply:
"Preliminary Draft Technical Report" PDTR is an old identifier stage/type abbreviation.

Right now we are always "TR CD" or "TR WD".

Reply:
Now I see what the problem is. There is an important misunderstanding here -- "PD" is *not* a stage. "PDTR" is an abbreviation, you cannot split it into "PD" and "TR". The same goes for "DIS" -- it is not "D" with "IS", it is just "DIS". This is logic that needs to be fixed throughout the gem.

Reply:
> Now I see what the problem is. There is an important misunderstanding here -- "PD" is _not_ a stage. "PDTR" is an abbreviation, you cannot split it into "PD" and "TR". The same goes for "DIS" -- it is not "D" with "IS", it is just "DIS". This is logic that needs to be fixed throughout the gem.

I don't use "PDTR", now it's rendered as "TR CD". As you said it's outdated.
"DIS" rendered as "DIS" only. But to render short versions of type+stage I use trick representing "DIS" as "D" and "FDIS" as "FD":
https://github.com/metanorma/pubid-iso/blob/0045e5f1bf0e0d252ce1f56599a41f099f7bbd00/lib/pubid/iso/renderer/base.rb#L32-L40
Then adding "TR" or "TS" stage.

Reply:
> I use trick representing "DIS" as "D" and "FDIS" as "FD":

Actually, this is kind of acceptable, and ISO internally they think this way as well.

> PDTR

The notion of "PDTR" just disappeared. I do not know how the "PD" stage fits into the harmonized code system.

> Is "ISO FDIS 1234" and "ISO FDTR 1234" are the same identifiers?
> From what I understand, "FDIS" means "Final Draft International Standard" but "FDTR" means "Final Draft Technical Report".

This is the question that you exactly should ask.

This is actually a very confusing issue (this is ISO's fault).

* In ISO, they have the stages 10 to 60. 50 is "Approval" stage, and they call 50.00 "Final text received or FDIS registered for formal approval".
* However, as you pointed out "FDIS" is "Final Draft International Standard"! How can a TR or TS document be a "FDIS"? That's why it does not make sense.  Similarly for "DIS" at stage 40.
* At stage 40 for a TR it would be "DTR", and we can "accept" that the phrase "DIS of a TR" => "DTR". And "DIS of a TS" => "DTS", "DIS of PAS" => "DPAS".
* At stage 50 for a TR it would be "FDTR", and we can "accept" that the phrase "FDIS of a TR" => "FDTR". And "FDIS of a TS" => "FDTS". A "PAS" cannot be at the "FDIS" stage (Stage 50), it would jump from stage 40 directly to 60.

So perhaps it is better to consider "FD" and "D" as the prefixes in this case.